apps/aecore/lib/aecore/pow/hashcash.ex

```diff
@@ -18,10 +20,43 @@ defmodule Aecore.Pow.Hashcash do
 
   @spec verify(binary(), non_neg_integer()) :: boolean()
   def verify(block_header_hash, difficulty) do
-    block_header_hash
-    |> Bits.extract()
-    |> Enum.take_while(fn bit -> bit == 0 end)
-    |> Enum.count() >= difficulty
+    {exp, significand} = Scientific.break_scientific(difficulty)
+    length = byte_size(block_header_hash)
+    zeros = 8 * max(0, length - exp)
+
+    case exp do
```

Comment:
This case looks weird, a cond is better

Reply:
I think it's better this way, there is no use switching to cond.

Reply:
This is literally a case transformed to be a cond, there is no matching on exp and you are just checking a bool expression, it's a trivial change so whatever